pkg/local_workflows/ignore_workflow/ignore_workflow.go

```diff
@@ -0,0 +1,333 @@
+	logger := invocationCtx.GetEnhancedLogger()
+	userInterface := invocationCtx.GetUserInterface()
+	config := invocationCtx.GetConfiguration()
+	id := invocationCtx.GetWorkflowIdentifier()
+
+	interactive := config.GetBool(InteractiveKey)
+	addCreateIgnoreDefaultConfigurationValues(invocationCtx, interactive)
+
+	userName, err := getUser(invocationCtx)
+	if err != nil {
+		return nil, fmt.Errorf("user is not authenticated: %w", err)
+	}
+
+	findingsId, err := config.GetStringWithError(FindingsIdKey)
+	if err != nil {
+		return nil, err
+	}
+
+	ignoreType, err := config.GetStringWithError(IgnoreTypeKey)
+	if err != nil {
+		return nil, err
+	}
+
+	repoUrl, err := config.GetStringWithError(code_workflow.ConfigurationRemoteRepoUrlFlagname)
+	if err != nil {
+		return nil, err
+	}
+
+	// read expiry time
+	// if expiration value is empty it will be treated as no-expire by the policy endpoint
+	expire, err := getExpireValue(config)
+	if err != nil {
+		return nil, err
+	}
+
+	reason, err := config.GetStringWithError(ReasonKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if interactive {
+		uiErr := userInterface.Output(fmt.Sprintf("👉🏼 Make sure the code containing the issue is committed, "+
+			"and pushed to a remote origin, so the approvers are able to analyze it.\n%s", getIgnoreRequestDetailsStructure(expire, userName, ignoreType)))
+		if uiErr != nil {
+			logger.Print(uiErr)
+		}
+	}
+
+	branchName, branchNameErr := git.BranchNameFromDir(config.GetString(configuration.INPUT_DIRECTORY))
+	if branchNameErr != nil {
+		logger.Print(branchNameErr)
+	}
+
+	payload := createPayload(repoUrl, branchName, expire, ignoreType, reason, findingsId)
+	response, err := sendCreateIgnore(invocationCtx, payload)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if interactive {
+		uiErr := userInterface.Output("\n✅ Your ignore request has been submitted.")
+		if uiErr != nil {
+			logger.Print(uiErr)
+		}
+	}
+
+	if config.GetBool(EnrichResponseKey) {
+		data, workflowDataErr := createIgnoreWorkflowData(
+			workflow.NewTypeIdentifier(id, ignoreCreateWorkflowName),
+			config,
+			policyResponseToSarifSuppression(response),
+			logger)
+		if workflowDataErr != nil {
+			return nil, workflowDataErr
+		}
+		output = append(output, data)
+	}
+
+	return output, err
+}
+
+func getIgnoreRequestDetailsStructure(expire *time.Time, userName string, ignoreType string) string {
+	requestedOn := time.Now().Format(time.DateOnly)
+	expireDisplayText := "Does not expire"
+	if expire != nil {
+		expireDisplayText = expire.Format(time.DateOnly)
+	}
+	return fmt.Sprintf("  Requested on:  %s\n  Requested by:  %s\n  Expiration:    %s\n  Ignore type:   %s", requestedOn, userName, expireDisplayText, ignoreType)
+}
+
+func getExpireValue(config configuration.Configuration) (*time.Time, error) {
+	shouldParse := config.IsSet(ExpirationKey) || config.GetBool(InteractiveKey)
+	if !shouldParse {
```

Comment:
nitpick: calling it `shouldNotExpire` and not checking for negation (i.e. `if shouldNotExpire {...}`), is easier to reason with